### PR TITLE
Add database migrations

### PR DESCRIPTION
**Pull request by Gabrielpanga**
Added database migrations.
Added automatic migrations on startup.
Added migrations cmd tool

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -4,6 +4,10 @@ SERVER_PARSER_EXTENDED=true
 SERVER_PARSER_LIMIT=2mb
 
 # Database
+DB_HOST=localhost
 DB_DATABASE=node-graphql
 DB_USER=postgres
 DB_PASSWORD=postgres
+
+# Other
+RUN_MIGRATIONS_ON_STARTUP=true
```
Comment (kaiomagalhaes) [inline .env.example]: please remove this variable

File: src/config/initializers/migrations.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import * as isEmpty from 'lodash.isempty';
```
Comment (kaiomagalhaes) [inline src/config/initializers/migrations.ts]: please remove the migrations from the initialize process

File: Makefile [hunk 3]
```diff
@@ -25,3 +25,8 @@ db-setup:
 
 db-setup-test:
   npm run build:tsc && NODE_ENV=test node scripts/db/setup.js
+	docker-compose up --remove-orphans -d db
```
Comment (kaiomagalhaes) [inline Makefile]: shouldn't we run the db before the npm build?

File: Makefile [hunk 4]
```diff
@@ -25,3 +25,8 @@ db-setup:
 
 db-setup-test:
   npm run build:tsc && NODE_ENV=test node scripts/db/setup.js
+	docker-compose up --remove-orphans -d db 
+
+dbmigration.create:
+	touch ./src/db/migrations/$(TODAY).$(name).up.sql
```
Comment (kaiomagalhaes) [inline Makefile]: this seems to have 4 spaces instead of 2

File: scripts/migrate.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
+const path = require('path');
```
Comment (kaiomagalhaes) [inline scripts/migrate.js]: this should be inside `scripts/db/`

File: scripts/migrate.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+const child_process = require('child_process');
```
Comment (kaiomagalhaes) [inline scripts/migrate.js]: we're using 2 spaces on the rest of the project

File: Makefile (hunk 3 above)
Comment (Gabrielpanga) [reply Makefile]: This is a bad merge, i removed this line since its on the top of the file.

File: Makefile (hunk 4 above)
Comment (Gabrielpanga) [reply Makefile]: When i opened the makefile i was breaking due to space indentation, i have to go back to the previous identation to make it work.

**Pull request merged**